learnosity_sdk/request/init.py

```diff
@@ -36,12 +36,11 @@ class Init(object):
     def __init__(
             self, service: str, security: Dict[str, Any], secret: str,
             request: Optional[Dict[str, Any]] = None, action:Optional[str] = None) -> None:
-        # Using None as a default value will throw mypy typecheck issues. This should be addressed
         self.service = service
         self.security = security.copy()
         self.secret = secret
         self.request = request
-        if hasattr(request, 'copy'):
+        if request is not None and hasattr(request, 'copy'):
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❔ question (non-blocking):** how do we currently behave if `request` is None? Is this a valid scenario or were we too defensive?

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously. -->
**🎯 suggestion:** maybe adding an explicit type for `self.request` might be useful as well for `mypy` and possibly the other attributes

Reply:
To be honest, I am unsure of the ramifications of `request` being None (i.e. how it is handled). I tried to make as minimal of a change as possible that did not change the current logic for that very reason.

Reply:
Now that we:
 - know the typing of `request`
 - found a bug related to handling the case where `request` is a string
 - opened a ticket to fix this bug
I think this comment can be marked as done.

Reply:
Sounds good 👍 Will mark as resolved and we can follow up later